unified_planning/io/python_writer.py

```diff
@@ -0,0 +1,332 @@
+        assert len(args) > 1
+        return f'emgr.And({", ".join(args)})'
+
+    def walk_or(self, expression, args):
+        assert len(args) > 1
+        return f'emgr.Or({", ".join(args)})'
+
+    def walk_not(self, expression, args):
+        assert len(args) == 1
+        return f'emgr.Not({args[0]})'
+
+    def walk_implies(self, expression, args):
+        assert len(args) == 2
+        return f'emgr.Implies({args[0]}, {args[1]})'
+
+    def walk_iff(self, expression, args):
+        assert len(args) == 2
+        return f'emgr.Iff({args[0]}, {args[1]})'
+
+    def walk_fluent_exp(self, expression, args):
+        fluent = expression.fluent()
+        if args:
+            return f'fluent_{fluent.name()}({", ".join(args)})'
+        return f'fluent_{fluent.name()}'
+
+    def walk_param_exp(self, expression, args):
+        assert len(args) == 0
+        p = expression.parameter()
+        return f'parameter_{p.name()}'
+
+    def walk_object_exp(self, expression, args):
+        assert len(args) == 0
+        o = expression.object()
+        return f'object_{o.name()}'
+
+    def walk_bool_constant(self, expression, args):
+        assert len(args) == 0
+        if expression.bool_constant_value():
+            return 'emgr.TRUE()'
+        else:
+            return 'emgr.FALSE()'
+
+    def walk_real_constant(self, expression, args):
+        assert len(args) == 0
+        return f'emgr.Real(Fraction({str(expression.real_constant_value().numerator)}, {str(expression.real_constant_value().denominator)}))'
+
+    def walk_int_constant(self, expression, args):
+        assert len(args) == 0
+        return f'emgr.Int({str(expression.constant_value())})'
+
+    def walk_plus(self, expression, args):
+        assert len(args) > 1
+        return f'emgr.Plus({", ".join(args)})'
+
+    def walk_minus(self, expression, args):
+        assert len(args) == 2
+        return f'emgr.Minus({args[0]}, {args[1]})'
+
+    def walk_times(self, expression, args):
+        assert len(args) > 1
+        return f'emgr.Times({", ".join(args)})'
+
+    def walk_div(self, expression, args):
+        assert len(args) == 2
+        return f'emgr.Div({args[0]}, {args[1]})'
+
+    def walk_le(self, expression, args):
+        assert len(args) == 2
+        return f'emgr.LE({args[0]}, {args[1]})'
+
+    def walk_lt(self, expression, args):
+        assert len(args) == 2
+        return f'emgr.LT({args[0]}, {args[1]})'
+
+    def walk_equals(self, expression, args):
+        assert len(args) == 2
+        return f'emgr.Equals({args[0]}, {args[1]})'
+
+
+class PythonWriter:
+    '''This class can be used to write a Problem in PDDL.'''
+
+    def __init__(self, problem: 'up.model.Problem'):
+        self.problem = problem
+
+    def _write_problem_code(self, out: IO[str]):
+
+        converter = ConverterToPythonString(self.problem.env)
+        out.write('import unified_planning as up\n')
+        out.write('from unified_planning.shortcuts import *\n')
```

Comment:
I would avoid this import, you should be able to create everything based on the `env` object and make everything reentrant, no? This makes the generated code much more reusable IMHO